vscode/vscode/vscode.ml

```diff
@@ -1810,39 +1812,254 @@ module Workspace = struct
+
+  val tooltip : t -> tooltip or_undefined [@@js.get]
+
+  val set_tooltip : t -> tooltip -> unit [@@js.set]
+
+  val collapsibleState : t -> TreeItemCollapsibleState.t or_undefined [@@js.get]
+
+  val set_collapsibleState : t -> TreeItemCollapsibleState.t -> unit [@@js.set]
+
+  val command : t -> Command.t or_undefined [@@js.get]
+
+  val set_command : t -> Command.t -> unit [@@js.set]
+
+  val contextValue : t -> string or_undefined [@@js.get]
+
+  val set_contextValue : t -> string -> unit [@@js.set]
+
+  val accessibilityInformation : t -> AccessibilityInformation.t or_undefined
+    [@@js.get]
+
+  val set_accessibilityInformation : t -> AccessibilityInformation.t -> unit
+    [@@js.set]
+end
+
+module TreeDataProvider = struct
+  type t = private (* interface *) Ojs.t [@@js]
+
+  val onDidChangeTreeData : t -> TreeItem.t or_undefined Event.t or_undefined
+    [@@js.get]
+
+  val getTreeItem : t -> element:TreeItem.t -> TreeItem.t Promise.t [@@js.call]
+
+  val getChildren :
+    t -> element:TreeItem.t or_undefined -> TreeItem.t list ProviderResult.t
+    [@@js.call]
+
+  val getParent :
+    t -> (element:TreeItem.t -> TreeItem.t ProviderResult.t) or_undefined
+    [@@js.call]
+
+  val resolveTreeItem :
+       t
+    -> (item:TreeItem.t -> element:TreeItem.t -> TreeItem.t ProviderResult.t)
+       or_undefined
+    [@@js.call]
+
+  val create :
+       getTreeItem:(element:TreeItem.t -> TreeItem.t Promise.t)
+    -> getChildren:
+         (element:TreeItem.t or_undefined -> TreeItem.t list ProviderResult.t)
+    -> ?getParent:(element:TreeItem.t -> TreeItem.t ProviderResult.t)
+    -> ?onDidChangeTreeData:TreeItem.t or_undefined Event.t
+    -> ?resolveTreeItem:
+         (item:TreeItem.t -> element:TreeItem.t -> TreeItem.t ProviderResult.t)
+    -> unit
+    -> t
+    [@@js.builder]
+end
+
+module TreeViewOptions = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  val treeDataProvider : t -> TreeDataProvider.t [@@js.get]
+
+  val showCollapseAll : t -> bool or_undefined [@@js.get]
+
+  val canSelectMany : t -> bool or_undefined [@@js.get]
+end
+
+module TreeView = struct
+  type t = private (* class *) Ojs.t [@@js]
+
+  val visible : t -> bool [@@js.get]
+
+  val message : t -> string or_undefined [@@js.get]
+
+  val title : t -> string or_undefined [@@js.get]
+
+  val description : t -> string or_undefined [@@js.get]
+
+  type revealOptions =
+    { select : bool or_undefined
+    ; focus : bool or_undefined
+    ; expand : bool or_undefined
+    }
+  [@@js]
+end
+
 module Window = struct
   val activeTextEditor : unit -> TextEditor.t or_undefined
     [@@js.get "vscode.window.activeTextEditor"]
 
-  val showQuickPick :
-       choices:QuickPickItem.t list
-    -> ?options:QuickPickOptions.t
-    -> ?token:CancellationToken.t
-    -> unit
-    -> QuickPickItem.t or_undefined Promise.t
-    [@@js.global "vscode.window.showQuickPick"]
+  val visibleTextEditors : unit -> TextEditor.t array
```

Comment:
We should prefer to use lists for the bindings since they're easier to manipulate in the extension code.

Reply:
Done, thanks!